src/DasMulli.Win32.ServiceUtils/Win32ServiceManager.cs

```diff
@@ -21,7 +21,7 @@ internal Win32ServiceManager(string machineName, string databaseName, INativeInt
             this.nativeInterop = nativeInterop;
         }
 
-        public void CreateService(string serviceName, string displayName, string description, string binaryPath, Win32ServiceCredentials credentials, bool autoStart = false, bool startImmediately = false, ErrorSeverity errorSeverity = ErrorSeverity.Normal)
+        public void CreateService(string serviceName, string displayName, string description, string binaryPath, Win32ServiceCredentials credentials, bool autoStart = false, bool startImmediately = false, ErrorSeverity errorSeverity = ErrorSeverity.Normal, ServiceFailureActions serviceFailureActions = null, bool failureActionsOnNonCrashFailures = false)
```

Comment:
A problem with adding parameters to public methods is that it is always a breaking change. Even if the argument is defaulted and therefor source code compatible, it is incompatible when updating built assemblies (e.g. via NuGet)..
This bugged me for a while and I've filed https://github.com/dasMulli/dotnet-win32-service/issues/20 for it, but did not get around to creating a "nice" solution.
For now, I'd suggest creating another method with the old signature that just forwards to the new method and remove the defaults.